hazelcast/src/main/java/com/hazelcast/internal/util/ExceptionUtil.java

```diff
@@ -118,7 +130,20 @@ public static RuntimeException peel(final Throwable t) {
         return exceptionWrapper.apply(t, message);
     }
 
-    public static RuntimeException rethrow(Throwable t) {
+    public static <W extends Throwable> Throwable wrapException(Throwable t, String message,
+                                                                BiFunction<Throwable, String, W> exceptionWrapper) {
+        if (t instanceof WrappableException) {
+            return ((WrappableException) t).wrap();
+        }
+        Throwable wrapped = tryWrapInSameClass(t);
```

Comment:
is all instances of `t` an HazelcastException for the client? if they all have the same base class, then for only the client side we may just make `HazelcastException` as `WrappableException `, and hence we may eliminate the need for the `tryWrapInSameClass ` method.

Reply:
This method is not for just the client. The only thing we know that `t` is a RuntimeException. There is no guarantee for it to extend HazelcastException.